master/internal/api_trials.go

```diff
@@ -311,6 +311,14 @@ func (a *apiServer) KillTrial(
 func (a *apiServer) GetExperimentTrials(
 	_ context.Context, req *apiv1.GetExperimentTrialsRequest,
 ) (*apiv1.GetExperimentTrialsResponse, error) {
+	ok, err := a.m.db.CheckExperimentExists(int(req.ExperimentId))
```

Comment:
It's weird that line 369 doesn't catch the experiment id. maybe fix that place?

Reply:
discussed offline, queryproto'ing a slice will never return not found so it's sort of impossible to fix that way without changing that.